devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -624,16 +628,34 @@ private void addProject(MavenDevModeLauncher.Builder builder, LocalProject local
         if (Files.isDirectory(testClassesDir)) {
             testClassesPath = testClassesDir.toAbsolutePath().toString();
         }
-        Path resourcesSourcesDir = localProject.getResourcesSourcesDir();
-        if (Files.isDirectory(resourcesSourcesDir)) {
-            resourcePath = resourcesSourcesDir.toAbsolutePath().toString();
-        }
-        Path testResourcesSourcesDir = localProject.getTestResourcesSourcesDir();
-        if (Files.isDirectory(testResourcesSourcesDir)) {
-            testResourcePath = testResourcesSourcesDir.toAbsolutePath().toString();
+        resourcePaths = localProject.getResourcesSourcesDirs().stream()
+                .filter(Files::isDirectory)
```

Comment:
I kept this filter to match with what was done before but do we rely need it?

Reply:
Maybe a better test could be `notExists || isDirectory`?

Reply:
I'm not 100% sure but that may have been the intention to filter out dirs that don't exist.

Reply:
If so it could be interesting to know why because in dev mode, you could want to add a new resource folder in the pom which would trigger an application restart then add a file in this new resource folder which won't be taken into account because of this filter (It is actually the scenario that I had in mind initially for the IT). Please also note that it seems to be inconsistent with Gradle according to what I understand from [this part of the code](https://github.com/quarkusio/quarkus/blob/main/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java#L427).

Reply:
When a POM changes we re-calculate the model anyway. I think it should still work.
The Gradle impl will need to be fixed.

Reply:
It would depend on the order, it would only work if you created the directory then added it to the pom.

Reply:
> It would depend on the order, it would only work if you created the directory then added it to the pom.

I confirm that it only works this way, if you modify the pom then create the directory it doesn't work that is why I was wondering if we could simply remove this filter because if we do it will work whatever the order.

Reply:
@aloubyansky so what do you want me to do finally ? Keep this filter as it is and add it to the gradle part or remove it?

Reply:
I guess if it doesn't break anything it can be removed. I guess a comment about this use-case could be useful.